apps/btshell/src/main.c

```diff
@@ -129,6 +129,21 @@ int btshell_full_disc_prev_chr_val;
 #define BTSHELL_AUTO_DEVICE_NAME    ""
 #endif
 
+#if MYNEWT_VAL(BLE_EXT_ADV)
+struct {
+    bool restart;
+    int conn_handle;
```

Comment:
make this `uint16_t` and use `BLE_HS_CONN_HANDLE_NONE` instead of `-1` for invalid handle